[DSI-8168] ApiSecret Encryption/Decryption

trickyward opened the pull request:
### Summary
Jira ticket: https://dfe-secureaccess.atlassian.net/browse/DSI-8168

---
### Changes

- Utilise `login.dfe.api-client/encryption` for encryption.
- Encrypt apiSecret before submitting the data to be persisted to the database.
- Resolve some package vulnerabilities with `npm audit fix`
- Updated some packages
- Updated some tests
---
### Related PRs

- https://github.com/DFE-Digital/login.dfe.api-client/pull/50
- https://github.com/DFE-Digital/login.dfe.public-api/pull/200
- https://github.com/DFE-Digital/login.dfe.manage/pull/331
- https://github.com/DFE-Digital/login.dfe.db-scripts/pull/2397

trickyward committed:
Add `login.dfe.api-client/encryption` for `apiSecret`

Test Suites: 101 passed, 101 total
Tests:       563 passed, 563 total
Snapshots:   0 total
Time:        8.014 s, estimated 13 s

trickyward committed:
Use login.dfe.api-client version 1.0.51-alpha.1

Test Suites: 101 passed, 101 total
Tests:       563 passed, 563 total
Snapshots:   0 total
Time:        9.325 s

trickyward committed:
Update package vulnerabilities with npm audit fix

Before: 5 vulnerabilities (4 low, 1 moderate)
After: 4 low severity vulnerabilities

Test Suites: 101 passed, 101 total
Tests:       563 passed, 563 total
Snapshots:   0 total
Time:        7.135 s, estimated 8 s

trickyward committed:
Update packages

Test Suites: 101 passed, 101 total
Tests:       563 passed, 563 total
Snapshots:   0 total
Time:        6.772 s

trickyward committed:
Merge branch 'main' into feature/dsi-8168-api-secret-encryption

ipredoi reviewed (commented):
Changes seem good to me 👍 
Ready to be approved after updating `login.dfe.api-client` package version

trickyward committed:
Updated login.dfe.api-client version

Test Suites: 100 passed, 100 total
Tests:       565 passed, 565 total
Snapshots:   0 total
Time:        10.938 s

trickyward committed:
Merge branch 'main' into feature/dsi-8168-api-secret-encryption

trickyward committed:
Use login.dfe.api-client 1.0.54

Test Suites: 100 passed, 100 total
Tests:       565 passed, 565 total
Snapshots:   0 total
Time:        9.663 s

trickyward committed:
Resolve some npm package vulnerabilities

Before: 5 vulnerabilities (4 low, 1 moderate)
After: 4 low severity vulnerabilities

Test Suites: 100 passed, 100 total
Tests:       565 passed, 565 total
Snapshots:   0 total
Time:        4.578 s, estimated 8 s

ipredoi reviewed (approved):
Looks good to me 👍
